bundles/org.openhab.binding.smarther/src/main/java/org/openhab/binding/smarther/internal/account/SmartherAuthorizationServlet.java

```diff
@@ -0,0 +1,273 @@
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.util.MultiMap;
+import org.eclipse.jetty.util.UrlEncoded;
+import org.openhab.binding.smarther.internal.api.dto.Location;
+import org.openhab.binding.smarther.internal.api.exception.SmartherGatewayException;
+import org.openhab.binding.smarther.internal.util.StringUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@code SmartherAuthorizationServlet} class acts as the registered endpoint for the user to automatically manage
+ * the BTicino/Legrand API authorization process.
+ * The servlet follows the OAuth2 Authorization Code flow, saving the resulting refreshToken within the Smarther Bridge.
+ *
+ * @author Fabio Possieri - Initial contribution
+ */
+@NonNullByDefault
+public class SmartherAuthorizationServlet extends HttpServlet {
+
+    private static final long serialVersionUID = 5199173744807168342L;
+
+    private static final String CONTENT_TYPE = "text/html;charset=UTF-8";
+    private static final String X_FORWARDED_PROTO = "X-Forwarded-Proto";
+
+    // Http request parameters
+    private static final String PARAM_CODE = "code";
+    private static final String PARAM_STATE = "state";
+    private static final String PARAM_ERROR = "error";
+
+    // Simple HTML templates for inserting messages.
+    private static final String HTML_EMPTY_APPLICATIONS = "<p class='block'>Manually add a Smarther Bridge to authorize it here<p>";
+    private static final String HTML_BRIDGE_AUTHORIZED = "<p class='block authorized'>Bridge authorized for Client Id %s</p>";
+    private static final String HTML_ERROR = "<p class='block error'>Call to Smarther API gateway failed with error: %s</p>";
+
+    private static final Pattern MESSAGE_KEY_PATTERN = Pattern.compile("\\$\\{([^\\}]+)\\}");
+
+    // Keys present in the index.html
+    private static final String KEY_PAGE_REFRESH = "pageRefresh";
+    private static final String HTML_META_REFRESH_CONTENT = "<meta http-equiv='refresh' content='10; url=%s'>";
+    private static final String KEY_AUTHORIZED_BRIDGE = "authorizedBridge";
+    private static final String KEY_ERROR = "error";
+    private static final String KEY_APPLICATIONS = "applications";
+    private static final String KEY_REDIRECT_URI = "redirectUri";
+    // Keys present in the application.html
+    private static final String APPLICATION_ID = "application.id";
+    private static final String APPLICATION_NAME = "application.name";
+    private static final String APPLICATION_LOCATIONS = "application.locations";
+    private static final String APPLICATION_AUTHORIZED_CLASS = "application.authorized";
+    private static final String APPLICATION_AUTHORIZE = "application.authorize";
+
+    private final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
```

Comment:
here too

Reply:
I think I've used it everywhere I needed to have a logger instance :)